spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -145,6 +147,10 @@ public List<ServiceInstance> getInstances(String serviceId) {
 		}
 		final String primaryPortName = discoveredPrimaryPortName.orElse(this.properties.getPrimaryPortName());
 
+		// reading secured value from svcMetadata, so we can read from service annotations
+		// and labels both (as per documentation)
+		boolean secured = svcMetadata.containsKey(SECURED_KEY) ? Boolean.valueOf(svcMetadata.get(SECURED_KEY)) : false;
```

Comment:
Should we check the the labels and annotations directly?  The user may not add the labels and/or annotations to the service metadata

Reply:
Labels and annotation are added eventually to the svcMetadata just few lines up. I think it is not necessary to check again. If present than take values otherwise is considered false.

Reply:
But labels and annotations may not be added, you can disable it via properties

Reply:
I've got your point. I'm going to fix it.

Thanks

Reply:
done.